### ISSUE TITLE
Movie files with metadata are not populating the database descriptions

### ISSUE THREAD

**Issue by andrewbird** [MythTV/mythtv#302]
I have `.mp4` files obtained from the BBC using get_iplayer. They have metadata tags describing the contents. Here is the output from Atomicparsley.

~~~
# atomicparsley The_Friday_Documentary_-_The_Mini_Is_30_p041yqmt_original.mp4 -t

Atom "stik" contains: TV Show
Atom "cprt" contains: 2020 British Broadcasting Corporation, all rights reserved
Atom "©nam" contains: The Mini Is 30
Atom "©ART" contains: BBC
Atom "aART" contains: BBC TV
Atom "©alb" contains: The Friday Documentary
Atom "©grp" contains: Factual,Documentaries,Cars & Motors
Atom "©wrt" contains: BBC iPlayer
Atom "©gen" contains: Factual
Atom "©cmt" contains: Documentary from 1989. The world's most famous small car celebrates its 30th birthday.
Atom "©day" contains: 1989-08-23T17:00:00+01:00
Atom "hdvd" contains: 1
Atom "tvsh" contains: The Friday Documentary
Atom "tven" contains: 198908231700
Atom "tvnn" contains: BBC
Atom "desc" contains: Documentary from 1989. The world's most famous small car celebrates its 30th birthday.
Atom "ldes" contains: The world's most famous small car celebrates its 30th birthday. Mini enthusiasts around the world, including Spike Milligan and Tony Benn, explain the little car's unique character. (1989)

PLAY: https://www.bbc.co.uk/iplayer/episode/p041yqmt

INFO: https://www.bbc.co.uk/programmes/p041yqmt
Atom "©lyr" contains: The world's most famous small car celebrates its 30th birthday. Mini enthusiasts around the world, including Spike Milligan and Tony Benn, explain the little car's unique character. (1989)

PLAY: https://www.bbc.co.uk/iplayer/episode/p041yqmt

INFO: https://www.bbc.co.uk/programmes/p041yqmt
Atom "covr" contains: 1 piece of artwork
~~~

I would have expected that info to be used to populate the mythtv database, but it seems that it's being thrown away and replaced by searching the filename in some internet database. This is close to certain failure as the filename comes straight out the BBC Iplayer platform and doesn't seem to be particularly helpful. Here's the output from the API's
`/Content/GetVideoList` that I see after `mythutil  --scanvideos` has been run.

~~~
# xmllint --format /tmp/a.xml
<?xml version="1.0" encoding="UTF-8"?>
<VideoMetadataInfoList xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="1.02" serializerVersion="1.1">
  <StartIndex>12</StartIndex>
  <Count>1</Count>
  <CurrentPage>13</CurrentPage>
  <TotalPages>31</TotalPages>
  <TotalAvailable>31</TotalAvailable>
  <AsOf>2021-01-18T18:41:16Z</AsOf>
  <Version>30.20181231-1</Version>
  <ProtoVer>91</ProtoVer>
  <VideoMetadataInfos>
    <VideoMetadataInfo>
      <Id>925</Id>
      <Title>The Friday Documentary - The Mini Is 30 p041yqmt original</Title>
      <SubTitle/>
      <Tagline/>
      <Director>Unknown</Director>
      <Studio/>
      <Description>None</Description>
      <Certification>NR</Certification>
      <Inetref>00000000</Inetref>
      <Collectionref>-1</Collectionref>
      <HomePage/>
      <ReleaseDate/>
      <AddDate>2020-09-10T23:00:00Z</AddDate>
      <UserRating>0</UserRating>
      <ChildID>-1</ChildID>
      <Length>0</Length>
      <PlayCount>0</PlayCount>
      <Season>0</Season>
      <Episode>0</Episode>
      <ParentalLevel>1</ParentalLevel>
      <Visible>true</Visible>
      <Watched>false</Watched>
      <Processed>false</Processed>
      <ContentType>MOVIE</ContentType>
      <FileName>The_Friday_Documentary_-_The_Mini_Is_30_p041yqmt_original.mp4</FileName>
      <Hash>ca64e3808c425f22</Hash>
      <HostName>oliver</HostName>
      <Coverart/>
      <Fanart/>
      <Banner/>
      <Screenshot/>
      <Trailer/>
      <Artwork>
        <ArtworkInfos/>
      </Artwork>
      <Cast>
        <CastMembers/>
      </Cast>
      <Genres>
        <GenreList/>
      </Genres>
    </VideoMetadataInfo>
  </VideoMetadataInfos>
</VideoMetadataInfoList>
~~~

As you can see the only pertinent info is the filename. Sometime I get cover art for some totally unrelated thing.

* **Platform**: Ubuntu 16.04

* **MythTV version**: 30.0

* **Package version**: 2:30.0+fixes.202008030353.622c949~ubuntu16.04.1 amd64 

* **Component**: backend

### What steps will reproduce the bug?
It seems to be normal operation

### How often does it reproduce? Is there a required condition?
Always

### What is the expected behaviour?
To populate the database from the metadata tagging on the `.mp4` file

Edited: better formatting for xml

**Comment by andrewbird** [comment]
I do have python amongst my skills, and i'm not averse to doing a little work on this. Is there a place that mythutil --scanvideos looks for scripts that I can plug in something to convert file metadata that would be ideal?

**Comment by andrewbird** [comment]
I see the place `mythtv/mythtv/programs/scripts/metadata/Movie/` where a metadata script might live. What determines which script gets run on `mythutil --scanvideos` and what arguments it gets called with, as the existing script `tmdb3.py` doesn't seem to have knowledge of the filename, only keywords?

**Comment by andrewbird** [comment]
Okay so I saw this PR #303 which seems to give me the answers I'm looking for.

**Comment by andrewbird** [comment]
So I'm making some headway with the grabber,  I have it parsing the files and generating the XML. I need to get the filename passed into it, but at present I only see args for searching that seem to be components of the filename. Is there any way I can control what options and args are passed to the grabber from the backend?

**Comment by rcrdnalor** [comment]
>>  Is there any way I can control what options and args are passed to the grabber from the backend?

No, there is not.
The -M, -D, -C options are hard coded in MythTV.
See
https://www.mythtv.org/wiki/MythTV_Universal_Metadata_Format

If you want to have an easy and fast solution, then I propose to use an User-Job.
See the MythTV wiki at 
https://www.mythtv.org/wiki/User_Jobs
This user-job can communicate with the MythTV's Services-API, see
https://www.mythtv.org/wiki/Services_API 
https://forum.mythtv.org/viewforum.php?f=33
and update the metadata stored in the database.
The forum welcomes you to start a discussion at 
https://forum.mythtv.org/viewforum.php?f=2

An other option, is to write a plugin:
See 
https://forum.mythtv.org/viewforum.php?f=6
https://www.mythtv.org/wiki/Building_Plugins:HelloMyth

Last, but not least, you may add additional functionality to the
MythTV project by proposing a concept and good architecture
to the full featured C++ core of mythtv.
Your feature will be likely placed in the libmythtv libraries and
used by the backend, frontend and command line utilities.
This concept needs agreement with the developers.
Of course, this needs to be accomplished by a pull-request on github.


In other words,
I have not told you anything new, everything is already documented.

I close this issue by now.
You may re-open it or create a new one, in case you decide
to implement the last offered option.
As I said, contributions to the MythTV project are always welcome.
